template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java

```diff
@@ -45,12 +47,17 @@
 
         switch (roleType) {
             case "RANGER_ADMIN":
-                RdsView rangerRdsView = new RdsView(getFirstRDSConfigOptional(templatePreparationObject).get());
-                variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_host").value(rangerRdsView.getHost()));
-                variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_name").value(rangerRdsView.getDatabaseName()));
-                variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_type").value(rangerRdsView.getSubprotocol()));
-                variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_user").value(rangerRdsView.getConnectionUserName()));
-                variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_password").value(rangerRdsView.getConnectionPassword()));
+                RDSConfig rdsConfig = getRDSConfig(templatePreparationObject);
+                RdsView rangerRdsView = new RdsView(rdsConfig);
+                if (rangerRdsView.getDatabaseVendor() == DatabaseVendor.POSTGRES) {
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_host").value(rangerRdsView.getHost()));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_name").value(rangerRdsView.getDatabaseName()));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_type").value("PostgreSQL"));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_user").value(rangerRdsView.getConnectionUserName()));
+                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_password").value(rangerRdsView.getConnectionPassword()));
+                } else {
+                    throw new CloudbreakServiceException("Ranger database supports only PostgreSQL");
+                }
                 break;
```

Comment:
I think database types supported for Ranger is configurable in SupportedDatabaseProvider.

Reply:
Right, so if that code is correct then it shall never reach this point.

 Unfortunately, SupportedDatabaseProvider is combined for Ambari an CM, and CM supports only PostgreSQL